### DISCUSSION TITLE
等宽字体的判断逻辑是什么，用户自行安装的等宽字体无法在v23被识别

### DISCUSSION THREAD

**Discussion by MomentDerek** [linuxdeepin/developer-center#4919] [Troubleshooting | 故障排查]
![image](https://github.com/linuxdeepin/developer-center/assets/40252940/6019a8f9-ce60-4675-a9fd-d5d160ed37f3)

![image](https://github.com/linuxdeepin/developer-center/assets/40252940/bbac7b70-c7b5-41e4-ad41-6b88a5bf5879)

记得在升级v23前，在20.9都是能被识别的？
比如下面地址中的字体包，经尝试，都无法识别，反而在Intellij IDEA中能正确识别为等宽字体
https://bbs.deepin.org/post/251539
或者是github中直接下载的等距更纱黑体[Sarasa-Mono-SC](https://github.com/be5invis/Sarasa-Gothic)
或者是相关的封装字体https://github.com/XuanXiaoming/Sarasa-Mono-SC-Nerd

**Comment by BLumia** [comment]
印象里之前看到说法是 daemon 会按照用户语言过滤字体 （ https://github.com/linuxdeepin/developer-center/issues/3392#issuecomment-1449541508 ）。这个行为是不是可以考虑去掉或者改成可配置的？cc @Decodetalkers @meiyixiang

**Comment by Decodetalkers** [comment]
你试试设置成英文环境应该就出来了

**Reply by MomentDerek** [reply]
我多试了试字体，大概判断是会去读取字体族的名称，过滤出包含mono这个单词等字体族，但是我下载的字体等字体族是通过短杠“-”来连接单词的，所以无法识别为等宽字体，把字体族中的短杠换为空格似乎就解决问题了

**Comment by Decodetalkers** [comment]
有啊，sara mono J

等下，你展示的不是nerd字体吗？

**Reply by MomentDerek** [reply]
是的，但是字体信息里不是用空格间隔，而是用短杠连接的单词，无法正确地被分词被识别

**Reply by Decodetalkers** [reply]
你答非所问啊，我说你用的nerd字体没有标记mono,所以没有，而且你给的github release也没有这个nerd字体，它里面其他的mono字体我都可以拿到的，你这个nerd字体不是mono吧，而且不建议用pua字体作为系统字体，编辑器使用下差不多了

**Reply by MomentDerek** [reply]
是mono字体，只不过字体族的名称是Sarasa-Mono-SC-Nerd，而不是Sarasa Mono SC Nerd，所以无法识别

**Reply by MomentDerek** [reply]
改名后重新打包后就没问题了

**Comment by ArchieMeng** [comment]
相关实现commit是https://github.com/linuxdeepin/deepin-terminal/commit/c569f1855135ac6489de2d3ee399a94a3e6c226d
理论上名字带Mono（无论大小写）的都会被识别成等宽字体。你可以看看deepin-terminal的Log中，打印这个字体的时候输出的信息。log位置为~/.cache/deepin/deepin-terminal/deepin-terminal.log

**Reply by BLumia** [reply]
和终端关系不太大吧，这个截图是控制中心的，应该用的是 daemon 提供的结果

**Reply by ArchieMeng** [reply]
哦哦。看错了。
